### PR TITLE
refactor : separate the different transaction pattern processing logic

### PR DESCRIPTION
**Pull request by ph3636**

### CODE REVIEW

**Comment by long187** [comment]
Are all three models regression tested with unit tests?

**Comment by ph3636** [comment]
> Are all three models regression tested with unit tests?

Yes, I can add more if it's not enough

**Comment by ph3636** [comment]
> I have cloned you branch and tested SagaCore. It works OK

ok,thanks

File: server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java [hunk 1]
```diff
@@ -70,20 +54,21 @@
...
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import static io.seata.core.exception.TransactionExceptionCode.FailedToSendBranchCommitRequest;
-import static io.seata.core.exception.TransactionExceptionCode.FailedToSendBranchRollbackRequest;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
```
Comment (lzf971107) [inline server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java]: Hello, plase keep jdk related import classes at the forefront.

File: server/src/main/java/io/seata/server/transaction/saga/SagaCore.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+import io.seata.server.coordinator.AbstractCore;
+import io.seata.server.session.BranchSession;
+import io.seata.server.session.GlobalSession;
+import io.seata.server.session.SessionHelper;
+import io.seata.server.session.SessionHolder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
```
Comment (lzf971107) [inline server/src/main/java/io/seata/server/transaction/saga/SagaCore.java]: Hello, plase keep jdk related import classes at the forefront.

File: server/src/main/java/io/seata/server/coordinator/AbstractCore.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+import io.seata.server.lock.LockerFactory;
+import io.seata.server.session.BranchSession;
+import io.seata.server.session.GlobalSession;
+import io.seata.server.session.SessionHelper;
+import io.seata.server.session.SessionHolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
```
Comment (lzf971107) [inline server/src/main/java/io/seata/server/coordinator/AbstractCore.java]: Hello, plase keep jdk related import classes at the forefront.

File: server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java (hunk 1 above)
Comment (ph3636) [reply server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java]: thanks, done

File: server/src/main/java/io/seata/server/transaction/saga/SagaCore.java (hunk 2 above)
Comment (ph3636) [reply server/src/main/java/io/seata/server/transaction/saga/SagaCore.java]: thanks, done

File: server/src/main/java/io/seata/server/coordinator/AbstractCore.java (hunk 3 above)
Comment (ph3636) [reply server/src/main/java/io/seata/server/coordinator/AbstractCore.java]: thanks, done

**Comment by zjinlei** [comment]
please resolve conflicts

File: server/src/main/java/io/seata/server/coordinator/Core.java [hunk 4]
```diff
@@ -29,27 +31,54 @@
...
+    /**
+     * Commit a branch transaction.
+     *
+     * @param globalSession the global session
+     * @param branchSession the branch session
+     * @return Status of the branch after committing.
+     * @throws TransactionException Any exception that fails this will be wrapped with TransactionException and thrown
+     *                              out.
+     */
+    BranchStatus branchCommit(GlobalSession globalSession, BranchSession branchSession) throws TransactionException;
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/coordinator/Core.java]: why not impl ResourceManagerInbound？

**Comment by slievrly** [comment]
@ph3636 DefaultCoordinatorTest test failed.

File: server/src/main/java/io/seata/server/coordinator/AbstractCore.java [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+    protected LockManager lockManager = LockerFactory.getLockManager();
+
+    protected ServerMessageSender messageSender;
+
+    public AbstractCore(ServerMessageSender messageSender) {
+        this.messageSender = messageSender;
+    }
+
+    public abstract BranchType getBranchType();
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/coordinator/AbstractCore.java]: getHandleBranchType

**Comment by ph3636** [comment]
> @ph3636 DefaultCoordinatorTest test failed.

done

**Pull request merged**